live-build/misc/migration-scripts/dx_apply

```diff
@@ -16,9 +16,190 @@
+#
+TMPDIR=$(mktemp -d -p "/tmp" -t delphix.XXXXXXX)
+FSNAME=$(basename "$TMPDIR")
+zfs create -o canmount=off -o mountpoint=none "$RPOOL/ROOT"
+zfs create -o canmount=noauto -o mountpoint=/ "$RPOOL/ROOT/$FSNAME"
+
+#
+# Temporarily set the mountpoint to point to a temporary
+# root directory, and unpack our cpio archive to it.
+#
+TMP_ROOT="$TMPDIR/root"
+zfs set mountpoint="$TMP_ROOT" "$RPOOL/ROOT/$FSNAME"
+zfs mount "$RPOOL/ROOT/$FSNAME"
+(
+	cd "$TMP_ROOT"
+	cpio -imu
+) <"$ARCHIVE_DIR/os-root.cpio" 2>&1 ||
+	die "failed to unpack os-root.cpio"
+
+#
+# WIP: Add here any operations that generate files from illumos,
+#	that we may want to carry over and use in Linux. Examples
+#	include networking interface configuration.
+#
+
+#
+# Set things up for the FreeBSD bootloader to provide an option for
+# booting into the Linux environment This happens in two steps:
+# [1] We place vmlinuz and initrd from our migration image into
+#     the /boot directory of the current system, so the bootloader
+#     can load them.
+# [2] Add a new option in the bootloader's menu for Linux. This is
+#     done by appending the configuration of the new option in our
+#     current menu file.
+#
+VMLINUZ=$(find "$TMP_ROOT/boot/vmlinuz-"* | sort -r | head -n 1)
+INITRD_IMG=$(find "$TMP_ROOT/boot/initrd.img-"* | sort -r | head -n 1)
+cp "$VMLINUZ" /boot
+cp "$INITRD_IMG" /boot
+
+#
+# This command loads the Linux Kernel Compressed Executable in memory
+# from the OK prompt of the FreeBSD bootloader.
+#
+# - We specify the root filesystem type to be ZFS and its root dataset
+#   the one that we just created for the migration image.
+#
+# - We set the console parameter during load twice. Once for each
+#   console technology:
+#      tty0 - Makes kernel messages appear in the first virtual
+#             terminal.
+#      ttyS0 - Makes kernel messages appear in the frst serial
+#             port. The 115200n8 part means the serial connection
+#             is made at 115200 baud 8n1.
+#
+# - We specify zfsforce=1 because otherwise we would be failing at
+#   pool import because of ZFS's hostname check.
+#
+LOAD_VMLINUZ_OK_ARG=(
+	"/boot/$(basename "$VMLINUZ")"
+	"root=ZFS=$RPOOL/ROOT/$FSNAME"
+	'console=tty0 console=ttyS0,115200n8'
+	'zfsforce=1'
+)
+LOAD_VMLINUZ_OK_CMD="load ${LOAD_VMLINUZ_OK_ARG[*]}"
+
+#
+# This command loads the initial RAM disk as the initial root filessytem
+# which is needed as part of the booting process.
+#
+LOAD_INITRD_OK_CMD="load -t rootfs /boot/$(basename "$INITRD_IMG")"
+
+#
+# This is the command to be issued whenever our new option in the
+# bootloader menu is chosen. It is FICL code (a Forth-inspired
+# command language) that is basically read as:
+# 	arg func; arg func; func.
+# In our case we:
+# [1] Evaluate the command to load vmlinuz
+# [2] Evaluate the command to load initrd
+# [3] boot (with the above two loaded)
+#
+MAIN_MENU_FICL=(
+	's\"' $LOAD_VMLINUZ_OK_CMD'\"' evaluate
+	's\"' $LOAD_INITRD_OK_CMD'\"' evaluate
```

Comment:
Not sure if I have a different version of `shellcheck` on my mac from the one run by the PR automation, but while the PR checks seem to all pass, mine complain about the escaping in those 2 lines:
```
In live-build/misc/migration-scripts/dx_apply line 171:
	's\"' $LOAD_VMLINUZ_OK_CMD'\"' evaluate
              ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.


In live-build/misc/migration-scripts/dx_apply line 172:
	's\"' $LOAD_INITRD_OK_CMD'\"' evaluate
              ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
```

Reply:
I changed that to have a single whole string per argument with double quotes.
The only awkward thing is that if you want something like `\"` you need to escape both the slash and the quote, leading in -> `\\\"`. After looking at this for a while that seemed nicer to me than mixing and matching single and double quotes.